src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -2249,5 +2254,127 @@ public final Object removeTransientAttribute(final Object key) {
         if (this.transientAttributes != null) return this.transientAttributes.remove(key);
         else return null;
     }
-}
 
+    /**
+     * Reverse-complement bases and reverse quality scores along with known optional attributes that
+     * need the same treatment. Changes made after making a copy of the bases, qualities,
+     * and attributes. If in-place update is needed use {@link #reverseComplement(SAMRecord, boolean)}.
+     * See {@link #TAGS_TO_REVERSE_COMPLEMENT} {@link #TAGS_TO_REVERSE}
+     * for the default set of tags that are handled.
+     */
+    public static void reverseComplement(final SAMRecord rec) {
+        reverseComplement(rec, false);
+    }
+
+    /**
+     * Reverse-complement bases and reverse quality scores along with known optional attributes that
+     * need the same treatment. Optionally makes a copy of the bases, qualities or attributes instead
+     * of altering them in-place. See {@link #TAGS_TO_REVERSE_COMPLEMENT} {@link #TAGS_TO_REVERSE}
+     * for the default set of tags that are handled.
+     *
+     * @param rec Record to reverse complement.
+     * @param inplace Setting this to false will clone all attributes, bases and qualities before changing the values.
+     */
+    public static void reverseComplement(final SAMRecord rec, boolean inplace) {
```

Comment:
similarly here, shouldn't it be non-static and only take inplace as an argument?